### PR TITLE
Added sublist search

### PR DESCRIPTION
**Pull request by Pratham-123**
### **Describe your change:**

Added sublist search

* [x] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [x] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: searches/recursive_substring_search.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
+# Recursive Python3 program to find if a given pattern is 
+# present in a text 
+
+def exactMatch(text, pat, text_index, pat_index):
```
Comment (taseikyo) [inline searches/recursive_substring_search.py:4]: The function name should be snake_case as described in CONTRIBUTING.md.
```suggestion
def exact_match(text, pat, text_index, pat_index): 
```
Type hints  and doctests are also required :)

File: searches/sublist_search.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
+def is_Sublist(l, s):
```
Comment (taseikyo) [inline searches/sublist_search.py:1]:
```suggestion
def is_sublist(l, s):
```

File: searches/recursive_substring_search.py (hunk 1 above)
Comment (Pratham-123) [reply searches/recursive_substring_search.py]: done

File: searches/sublist_search.py (hunk 2 above)
Comment (Pratham-123) [reply searches/sublist_search.py]: done

**Comment by NumberPiOso** [comment]
You clearly checked multiple boxes without fulfilling the requirements.

You should have done two pull requests, one for each file.

These should contain doctests and typehints for every function.

**Comment by xcodz-dot** [comment]
Follow these steps in order to go good:
* Remove either of your files (you can contribute your second file in a separate PR)
* run these commands and go through the CONTRIBUTING.md again (its updated with help instructions)
  - `pip install pre-commit`
  - `pre-commit run -a` in the root of your local copy

**Comment by dhruvmanila** [comment]
@algorithms-keeper review

**Pull request closed without merging**